src/eq.js

```diff
@@ -19,5 +19,20 @@ var _curry2 = require('./internal/_curry2');
  *      R.eq(o, {}); //=> false
  *      R.eq(1, 1); //=> true
  *      R.eq(1, '1'); //=> false
+ *      R.eq(0, -0); //=> false
+ *      R.eq(NaN, NaN); //=> true
  */
-module.exports = _curry2(function eq(a, b) { return a === b; });
+module.exports = _curry2(function eq(a, b) {
+    // Implementation comes from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Polyfill
+    if (!Object.is) {
+        if (a === 0 && b === 0) {
+            return 1 / a === 1 / b;
+        }
+        if (a !== a) {
+            return b !== b;
+        }
+        return a === b;
+    } else {
+        return Object.is(a, b);
+    }
+});
```

Comment:
Yup, it seems that they are equivalent, although I'm really not a big fan of the ternary operator, how about:
``` javascript
if (a === 0 && b === 0) {
    return 1 / a === 1 / b;
} else {
    return a === b || (a !== a && b != b);
}
```